docs/misc/state_witness_size_limits.md

```diff
@@ -0,0 +1,35 @@
+### State witness size limits
+
+Some limits were introduced to keep the size of `ChunkStateWitness` reasonable.
+`ChunkStateWitness` contains all the incoming transactions and receipts that will be processed during chunk application and in theory a single receipt could be tens of megabatytes in size. Distributing a `ChunkStateWitness` this large would be troublesome, so we limit the size and number
+of transactions, receipts, etc. The limits aim to keep the total uncompressed size of `ChunkStateWitness` under 16MiB.
+
+There are two types of size limits:
+* Hard limit - the size must be below this limit, anything else is considered invalid
+* Soft limit - things are added until the limit is exceeded, after that things stop being added. The last added thing is allowed to slightly exceed the limit.
+
+The limits are:
+* `max_transaction_size = 1.5 MiB`
+    * All transactions must be below 1.5 MiB, otherwise they'll be considered invalid and rejected.
+    * Previously was 4MiB, now reduced to 1.5MiB
+* `max_receipt_size = 4 MiB`:
+    * All receipts must be below 4 MiB, otherwise they'll be considered invalid and rejected.
+    * Previously there was no limit on receipt size. Set to 4MiB, might be reduced to 1.5MiB in the future to match the transaction limit.
+* `combined_transactions_size_limit = 2 MiB`
+    * Hard limit on total size of transactions from this and previous chunk. `ChunkStateWitness` contains transactions from two chunks, this limit applies to the sum of their sizes.
+* `new_transactions_validation_state_size_soft_limit - 500 kiB`
+    * Validating new transactions generates storage proof (recorded trie nodes), which has to be limited. Once transaction validation generates more storage proof than this limit, the chunk producer stops adding new transactions to the chunk.
+* `per_receipt_storage_proof_size_limit = 4 MB`
```

Comment:
Here we use `MB` and above it is`MiB`, just want to confirm that it is accurate

Reply:
Yeah it's:

```yaml
per_receipt_storage_proof_size_limit: {old: 999_999_999_999_999, new: 4_000_000}
main_storage_proof_size_soft_limit: {old: 16_000_000, new: 3_000_000}
```
and
```yaml
max_transaction_size: {old: 4_194_304, new: 1_572_864}
max_receipt_size: {old: 999_999_999_999_999, new: 4_194_304}
```

A bit messy, but :shrug: